bin/travis

```diff
@@ -46,5 +46,12 @@ sbt 'set scalazVersion in ThisBuild := System.getenv("SCALAZ_VERSION")' ++$TRAVI
 
 echo "Uploading codecov"
 if [[ $SBT_COMMAND = *";coverage"* ]]; then
-   bash <(curl -s https://codecov.io/bash)
+    # `env -u` to unset sensitive environment variables the Codecov script does not need to see
+    env \
```

Comment:
I used `env -u` instead of `unset` because it only affects the bash child process running codecov.  The environment variables remain available for later use within `bin/travis`.  I was nervous `unset` would break publishing to sonatype or gh-pages, especially if this rolls into `release-0.16.x` and `master` where the Travis setup has changed.

But it sounds like Ross has a better version in the works anyway.  I'll close this one.